eos-phone-home

```diff
@@ -148,6 +153,43 @@ class PhoneHome(object):
 
         return None
 
+    def _get_metrics(self):
+        permissions_file = '/etc/metrics/eos-metrics-permissions.conf'
+        valid_environments = ['production', 'dev', 'test']
+
+        enabled = False
+        uploading_enabled = True
+        environment = 'production'
+
+        try:
+            config = configparser.ConfigParser()
+            config.read(permissions_file)
+            enabled = config['global'].getboolean('enabled', enabled)
+            uploading_enabled = config['global'].getboolean('uploading_enabled',
+                uploading_enabled)
+            environment = config['global'].get('environment', environment)
+        except Exception as e:
+            print("Unable to read %s, assuming fallback values." %
+                (permissions_file, ), e, file=sys.stderr)
+
+        # the event recorder replaces invalid environments with 'test'
+        if environment not in valid_environments:
+            print('Found invalid metrics environment', environment,
+                file=sys.stderr)
+            environment = 'test'
+
+        return (enabled and uploading_enabled, environment)
+
+    def _get_metrics_enabled(self):
+        enabled = self._lookup_or_get_variable('metrics')[0]
```

Comment:
(Technically the user might reboot the machine before completing the FBE – should we either gate sending pings on at least one user account existing, or include a boolean field indicating whether the FBE has completed?)